File: Rio/Rio.Web/Modules/ServerTypes/Workspace/ExamForm.ts
import { StringEditor, TextAreaEditor, LookupEditor, IntegerEditor, DecimalEditor, ImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ExamForm {
    Code: StringEditor;
    Name: StringEditor;
    Description: TextAreaEditor;
    SheetTypeId: LookupEditor;
    TotalQuestions: IntegerEditor;
    TotalMarks: IntegerEditor;
    NegativeMarks: DecimalEditor;
    OptionsAvailable: IntegerEditor;
    ResultCriteria: StringEditor;
    QuestionPaper: ImageUploadEditor;
    ModelAnswer: ImageUploadEditor;
    SelectedTenant: LookupEditor;
}

export class ExamForm extends PrefixedContext {
    static formKey = 'Workspace.Exam';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamForm.init)  {
            ExamForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = LookupEditor;
            var w3 = IntegerEditor;
            var w4 = DecimalEditor;
            var w5 = ImageUploadEditor;

            initFormType(ExamForm, [
                'Code', w0,
                'Name', w0,
                'Description', w1,
                'SheetTypeId', w2,
                'TotalQuestions', w3,
                'TotalMarks', w3,
                'NegativeMarks', w4,
                'OptionsAvailable', w3,
                'ResultCriteria', w0,
                'QuestionPaper', w5,
                'ModelAnswer', w5,
                'SelectedTenant', w2
            ]);
        }
    }
}

File: Rio/Rio.Web/Modules/Workspace/Exam/ExamForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.Exam")]
    [BasedOnRow(typeof(ExamRow), CheckNames = true)]
    public class ExamForm
    {
        [HalfWidth]
        public string Code { get; set; }
        [HalfWidth]
        public string Name { get; set; }
        [TextAreaEditor(Rows = 5)]
        [HalfWidth]
        public string Description { get; set; }
        [HalfWidth]
        public int TotalQuestions { get; set; }
        [HalfWidth]
        public int TotalMarks { get; set; }
        [HalfWidth]
        public float NegativeMarks { get; set; }
        [HalfWidth]
        public short OptionsAvailable { get; set; }
        [HalfWidth]
        public string ResultCriteria { get; set; }
        
    }
}